spec/index.html

```diff
@@ -922,12 +955,23 @@ <h3>Algorithm</h3>
 
       <ol id="ca">
         <li id="ca.1">Create the <a>canonicalization state</a>.
+          If the <a>input dataset</a> is an N-Quads document,
+          parse that document into an dataset,
+          retaining any blank node identifiers used within that document
+          in the <a>input blank node identifier map</a>,
+          otherwise arbitrary identifiers are assigned for each
+          blank node.
           <details>
             <summary>Explanation</summary>
             <p>This has the effect of initializing the
               <a>blank node to quads map</a>,
               and the <a>hash to blank nodes map</a>,
               as well as instantiating a new <a>canonical issuer</a>.</p>
+            <p>After this algorithm completes,
+              the <a>input blank node identifier map</a> state
+              may be used to correlate blank node identifiers
+              used in the <a>input dataset</a> to those used
+              in the <a>normalized dataset</a>.</p>
```

Comment:
We need to expose the input blank node ID => canonical blank node ID ordered map from the canonical issuer as well -- as it's the easiest thing for some implementations to use:

```suggestion
            <p>After this algorithm completes,
              the <a>input blank node identifier map</a> state
              and / or <a>canonical issuer</a> may be used to
              correlate blank node identifiers used in the
              <a>input dataset</a> to those used
              in the <a>normalized dataset</a>.</p>
```

I see there's a comment from @gkellogg that perhaps we want to expose the canonical issuer (or its input bnode ID => output bnode ID) map in the normalized dataset instead of a input bnode => output bnode ID map ... and that may simplify this here without losing anything.

If the "input blank node identifier map" always maps the abstract bnodes in the input dataset (and, therefore the normalized dataset as well) to the input blank node IDs, then the canonical issuer's ordered map can:

* be used to map from input blank node IDs to output blank node IDs (this is a single step), and
* it can be used in conjunction with the "input blank node identifier map" to go, in two steps, from bnodes to input bnode IDs to output node IDs.